k9mail/src/main/java/com/fsck/k9/mailstore/LocalFolder.java

```diff
@@ -1355,7 +1356,7 @@ protected void saveMessage(SQLiteDatabase db, Message message, boolean copy, Map
             cv.put("reply_to_list", Address.pack(message.getReplyTo()));
             cv.put("attachment_count", attachmentCount);
             cv.put("internal_date", message.getInternalDate() == null
-                    ? System.currentTimeMillis() : message.getInternalDate().getTime());
+                    ? SystemClock.elapsedRealtime() : message.getInternalDate().getTime());
```

Comment:
Ditto. This will set the date of the email to the amount of time since the phone rebooted.